sourced/ml/algorithms/id_splitter/features.py

```diff
@@ -0,0 +1,124 @@
+import logging
+import string
+import tarfile
+import warnings
+
+import numpy as np
+from keras.preprocessing.sequence import pad_sequences
+from modelforge.progress_bar import progress_bar
+
+
+DEFAULT_MAX_IDENTIFIER_LEN = 40  # default max length of sequence
+PADDING = "post"  # add padding values after input
+
+# In the CSV file, columns 0,1,2 contain statistics about the identifier.
+CSV_IDENTIFIERS_COL = 3  # Column 3 contains the input identifier e.g. "FooBar".
+CSV_SPLIT_IDENTIFIERS_COL = 4  # Column 4 contains the identifier lowercase and spitted "foo bar".
+
+
+def read_identifiers(csv_path: str, use_header: bool=True,
+                     max_identifier_len: int=DEFAULT_MAX_IDENTIFIER_LEN,
+                     identifiers_col: int=CSV_IDENTIFIERS_COL,
+                     split_identifiers_col: int=CSV_SPLIT_IDENTIFIERS_COL,
+                     shuffle: bool=True):
+    """
+    Reads and filters too long identifiers from CSV file.
+    :param csv_path: path to the CSV file.
+    :param use_header: uses header as normal line (True) or treat as header line with column names.
+    :param max_identifier_len: maximum length of raw identifier. Skip identifier if longer.
+    :param identifiers_col: column in CSV file for the raw identifier.
+    :param split_identifiers_col: column in CSV file for the splitted identifier.
+    :param shuffle: indicates whether to reorder the list of identifiers at random after reading it
```

Comment:
missed dot at the end

Reply:
This was adressed even though the diff is not showing it